scripts/hostcfgd

```diff
@@ -230,15 +274,18 @@ class FeatureHandler(object):
     def handler(self, feature_name, op, feature_cfg):
         if not feature_cfg:
             syslog.syslog(syslog.LOG_INFO, "Deregistering feature {}".format(feature_name))
+            self.lock.acquire()
```

Comment:
I understand the complication of keeping single threaded design considering limiting abilities for injecting own selectables into ConfigDBConnector.listen(). On the other hand I see many acquire/release blocks that might be forgotten if new code comes in. Can the acquire lock happen once in handler() and released once it is done so that we have all other methods call under lock?